### PR TITLE
Improvements to Monitor Ingestion Javadocs

### PR DESCRIPTION
**Pull request by jairmyree**
Improvements to Monitor Ingestion Javadocs

### CODE REVIEW

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/models/LogsUploadOptions.java
Comment (jairmyree) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/models/LogsUploadOptions.java:20]: I had to add the default constructor due to Java 21's warnings against constructors without javadocs.

**Comment by jairmyree** [comment]
Since there was strong Javadoc documentation and a sample for each client method, I decided to rely on the method-level Javadoc samples rather than write additional samples. @g2vinay what do you think about that approach?

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClient.java:90]:
```suggestion
 *         {@link LogsIngestionClient#uploadWithResponse(String, String, BinaryData, RequestOptions) uploadWithResponse(String, String, BinaryData, RequestOptions)} - Uploads logs to a Log Analytics workspace with options to configure the HTTP request.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:29]:
```suggestion
 *     using Microsoft Entra ID. To authenticate with Microsoft Entra ID, create a
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java:49]:
```suggestion
 *     This client encapsulates REST API calls, used to send data to a Log Analytics workspace, into a set of asynchronous operations.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java:83]:
```suggestion
 *         {@link LogsIngestionAsyncClient#upload(String, String, Iterable) upload(String, String, Iterable)} - Uploads logs to a Log Analytics workspace.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java:86]:
```suggestion
 *         {@link LogsIngestionAsyncClient#upload(String, String, Iterable, LogsUploadOptions) upload(String, String, Iterable, LogsUploadOptions)} - Uploads logs to a Log Analytics workspace with options to configure the upload request.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java:89]:
```suggestion
 *         {@link LogsIngestionAsyncClient#uploadWithResponse(String, String, BinaryData, RequestOptions) uploadWithResponse(String, String, BinaryData, RequestOptions)} - Uploads logs to a Log Analytics workspace with options to configure the HTTP request.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClient.java:84]:
```suggestion
 *         {@link LogsIngestionClient#upload(String, String, Iterable) upload(String, String, Iterable)} - Uploads logs to a Log Analytics workspace.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClient.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClient.java:50]:
```suggestion
 *     This client encapsulates REST API calls, used to send data to a Log Analytics workspace, into a set of synchronous operations.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:36]:
```suggestion
 *     for more information. See {@link com.azure.monitor.ingestion.LogsIngestionClientBuilder LogIngestionClientBuilder} for more examples on authenticating a client.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:88]: Instead of listing the built-in tables here, link to https://learn.microsoft.com/azure/azure-monitor/logs/logs-ingestion-api-overview#supported-tables. The list is likely to change over time, and the doc I linked to should be the source of truth.

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:102]:
```suggestion
 *     <a href="https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/monitor/azure-monitor-query#readme">Azure Monitor Query client library</a>.
```

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:111]:
```suggestion
 *     {@link com.azure.monitor.ingestion.LogsIngestionAsyncClient}, respectively.
```

**Comment by jairmyree** [comment]
Waiting until monitor GA to merge

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClientBuilder.java
Comment (srnagar) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionClientBuilder.java:69]: Remove this line

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/models/LogsUploadOptions.java
Comment (srnagar) [reply sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/models/LogsUploadOptions.java:20]: Remove this line.

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (srnagar) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:44]: Using "Azure tables" here can be mistaken for [Azure Tables storage](https://learn.microsoft.com/en-us/azure/storage/tables/table-storage-overview)

File: sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/package-info.java:59]:
```suggestion
 *     The API allows you to send data to <a href="https://learn.microsoft.com/azure/azure-monitor/logs/logs-ingestion-api-overview#supported-tables">supported tables</a>
```

**Review by scottaddie — APPROVED** [review]

**Review by g2vinay — APPROVED** [review]

**Pull request merged**
